cpp/include/cudf/strings/convert/is_valid_fixed_point.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <cudf/column/column.hpp>
+
+namespace cudf {
+namespace strings {
+/**
+ * @addtogroup strings_convert
+ * @{
+ * @file
+ */
+
+/**
+ * @brief Returns a boolean column identifying strings in which all
+ * characters are valid for conversion to integers.
+ * 
+ * @param strings Strings instance for this operation.
+ * @param allow_decimal identification whether the format is decimal or not
+ * @param input_type input data type for check overflow
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column of boolean results for each string.
+ */
+std::unique_ptr<column> is_valid_fixed_point(
+  strings_column_view const& strings,
+  bool allow_decimal,
```

Comment:
I don't understand the purpose of this parameter anymore.

Reply:
The difference is `[0-9]` vs `[0-9](\.[0-9])?`.  if `allow_decimal` is true then we allow the number to be followed by a decimal point followed and more numbers.  If it is false we do not.

Reply:
I understood that much, I just don't understand why this is a desired option. In my mind this would be like having an `allow_decimal` flag to the `is_floating_point` API---both strike me as odd.

`bool` flags are always a code smell for me as they are warning sign of a single responsibility violation.

Reply:
This is to match what Spark and Hive do when casting strings to int values. Floating point allows for `NaN`, `-Inf`, etc that I guess we could special case yet again, but then the overflow wold not be properly checks, which is another feature that we are looking for.

Reply:
> This is to match what Spark and Hive do when casting strings to int values.

What behavior are you looking for? From my perspective, if `allow_decimal` is `False`, wouldn't that effectively make this function `is_valid_integer`?

Reply:
Yes if `allow_decimal` is `False` then the result would be equivalent to `is_valid_integer` and if you want two functions instead of one, that is fine. I am not hung up on a specific API. What I need is functionality to check for a specific pattern along with integer overflow.

Reply:
so , here we need 2 functions instead of 1 ?

Reply:
Yes, I think `is_valid_integer` should be distinct from `is_valid_fixed_point`. And ultimately we should just have [`is_valid_element`](https://github.com/rapidsai/cudf/pull/7094#discussion_r554180275).

Reply:
Based on what you have discussed, I think we should do this : 
@jrhemstad 
this pr is for  
```
unique_ptr<column> is_valid_integer(strings_column_view strings, data_type type);
```
and , i will file a new pr for 
```
unique_ptr<column> is_valid_fixed_point(strings_column_view strings, data_type type);
```

@revans2 
above api contains parameter `data_type` , that will help us to check overflow. 

while, for parameter `allow_decimal` requirements you mentioned  in #7080  , we can judge `allow_decimal` in spark-rapids,  if `allow_decimal` is true,  through JNI layer , we finally call `is_valid_fixed_point` in cudf cpp module, if `allow_decimal` is false , we finally call `is_valid_integer` in cudf cpp module, is it right ? in this way , `allow_decimal` parameter will not enter cudf. 

please give me some advice.

Reply:
For CUDF JNI I would rather mirror the underlying C++ API as much as possible. The Spark plugin can decide which CUDF API to call instead of pushing that into cudf.

Reply:
@jrhemstad I'm not sure `is_valid_element` will do what we want.  We are asking for two separate things, and perhaps it should be two separate API calls then.  First we want to check if the format of the string matches what we expect, next we want to check if when we parse the string as an integer if the number will overflow. In the current set of APIs `is_valid_fixed_point` and `is_valid_integer` the name of the function indicates the format of the string while the type we pass in, indicates the overflow value.  If we just have `is_valid_element` then we would need that same split in roles. Does the format match fixed point and we don't overflow parsing to a byte?  If that is too much for a single API to do, then perhaps what we want is to add `is_fixed_point` along with `is_integer` and `is_float`, and also add an API to check if a strings column would overflow when parsing to an integer of type X.